server/src/main/java/com/cloud/api/query/dao/TemplateJoinDaoImpl.java

```diff
@@ -522,6 +522,7 @@ public TemplateResponse newIsoResponse(TemplateJoinVO iso) {
         sc.setParameters("store_id", storeId);
         sc.setParameters("type", TemplateType.USER);
         sc.setParameters("templateState", VirtualMachineTemplate.State.Active);
+        sc.setParameters("public", Boolean.FALSE);
```

Comment:
@Pearl1594 why this change is needed?

Reply:
@shwstppr Deletion of image stores will be prevented only if there are private user templates present on the respective store.